### DISCUSSION TITLE
New score rank idea

### DISCUSSION THREAD

**Discussion by Givikap120** [ppy/osu#16998] [Ideas]
In old osu! we have A for high accuracy and S for high accuracy and no miss. There is a problem with this where 91% accuracy 30 miss run has A and also 99% accuracy 1 miss run has A.
In lazer we have 90-95% - A, and <100% S. This is fixes last problem, but I think no-miss (and slider break) run also must be awarded. So I think it will be good to add S+ rank to reward S rank with no miss/sliderbreak. Also, as alternative no miss S ranks will be SS, and 100% acc will be SSS.

**Comment by peppy** [comment]
Historically we had the "perfect" display for this purpose. Is that what you're after here? You're proposing combining that into the rank grades as well?

I'm not sure we want to change rank grades at this point in osu!'s history, as people know how they work and they are very easy to understand. Adding a new grade in between two existing ones is going to lead to confusion.

**Reply by Givikap120** [reply]
Perfect stands for the perfect combo, doesn't it? If you miss 1 slider-end you will not get "perfect" on the score screen.

In osu!lazer you already changed the rank system to be accuracy based instead of 300's based, and S requirements have been seriously reworked. I just said what 0 miss 0 slider-breaks (FC) is special for osu and should stand out from the default high accuracy scores.

Also, if you just add a "perfect" alternative to the FC, it will be a problem too. You need to "click" on the score to see this information. 

If new ranks are bad, what about an additional mark on a rank icon like in this image (silver for FC, golden for PC)? What do you think about this?

![image](https://user-images.githubusercontent.com/89256026/157249166-8cb19241-d7e6-4f99-a84d-464536cf6898.png)

**Reply by memesnightmares** [reply]
I agree that it would be beneficial to add some kind of distinction for FC plays as opposed to high accuracy plays that aren't FCs. Peppy, you say "I'm not sure we want to change rank grades at this point in osu!'s history, as people know how they work and they are very easy to understand" and I agree with that, but from what I understand (lazer only player) stable ranks are different than lazers as they are currently implemented. Because of that, I've heard confusion from stable players about why it's possible to get an S rank in lazer for a play that includes misses. My understanding is that most of lazers scoring is accuracy based which makes it feel a little redundant to me. I like the idea of ranks being some aggregate of different fields i.e. score and miss count which is how I understand it works on stable. Would you consider bringing lazers ranking system in-line with stables, since it sounds like the system there works well?

**Reply by monochrome22** [reply]
How would the marks be displayed on profile? 
Here: 
![profile grades](https://github.com/ppy/osu/assets/122969285/26bc4642-3c20-4eb2-b4cd-a8aa17a5343d)

If there's going to be a distinction between not FC, FC and PC, then it feels wrong to put them all under just "S" on profile. Putting them all separately would create too many badges on profile.

**Reply by ghost** [reply]
While I'm open to adding another grade or two, at the moment it all seems like it's just making the game more convoluted for the sake of "fixing" a problem that wouldn't be there in the first place if you just use Stable's system

**Comment by Biokax** [comment]
Adding S+, SS, and SSS for rewarding a full combo will be bad and make players confuse about the rank system.

Maybe a new filter that able players to see non full combo plays in local scores, since the A rank did not help to found them could be a great addition. If a new filter like this is added, it must have an option to hide 1-3 sliderend play for example because they are basically full combo

**Reply by Givikap120** [reply]
I was talking about S -> S+ -> SS or S -> SS -> SSS for High accuracy -> Full Combo -> 100% accuracy. 

If new ranks are bad what about additional marks (like ones in my peppy's answer).

**Reply by Biokax** [reply]
SSS and others are confusing and i'm not really with this idea, but the additional marks on your answer for full combo and perfect score are really interesting and I hope it can come to the game with some filters like "perfectscore=" and "fullcombo=" to display them for local scores in game

**Comment by iSimpFor** [comment]
i thought S+ is for S-ranks with vision-reducing mods?
that would very much confuse people just on that alone

i would prefer to add grades for accuracy to represent hitting the map to the song very well instead of for full combos because there is already a full combo marker, i'm not the dev however

**Reply by Givikap120** [reply]
Where is the full combo marker? There is a perfect combo marker, but not a full combo, doesn't it? Also, S+ = silver S is what just people came up with, the game does not tell what silver S is an S+.

**Reply by bdach** [reply]
> Also, S+ = silver S is what just people came up with, the game does not tell what silver S is an S+.

that's wrong, actually.

https://github.com/ppy/osu/blob/a352a140bc145f342fadc300797a2b05ade3f920/osu.Game/Scoring/ScoreRank.cs#L32-L34
https://github.com/ppy/osu/blob/a352a140bc145f342fadc300797a2b05ade3f920/osu.Game/Scoring/ScoreRank.cs#L40-L42

S+ and SS+ are used in lazer already for "silver" S/SS.

**Reply by Givikap120** [reply]
I have seen what silver S is SH in game files. Ok.
